### PR TITLE
2991 remove SFTP and ACF titan related codes

### LINKED ISSUE #2991: As tech lead, I need the sftp file transfer feature to be deprecated
**Description:**
At the end of May 2024, the legacy system (TDRS) will be decommissioned. This means that data files from STTs will no longer be processed in-parallel with processing in TDP, so the file transfer feature can be disabled. 

**Acceptance Criteria:** 

- [x] files submitted through TDP are not transferred to ACFTitan server. 
- [ ] file transfer env vars are no longer present in the codebase, circleci, or backend environments
- [x] backend logs do not show evidence of file transfer attempts. 
- [x] all other existing TDP features are functioning as expected. 
- [x] Testing Checklist has been run and all tests pass
- [x] README is updated, if necessary

**Tasks:**

- [x] Disable sftp file upload 
- [x] Delete sftp_task, remove the task and delete all references including _sftp_task.upload_ in data_file.views
- [x] Delete any associated upload task test (test_file_upload)
- [ ] ~~Delete the _LegacyFileTransfer_ model, and create migration to remove the data from app (Question: should we keep this model and all the history?)~~
- [x] Remove associated lines in readme files (File-transfer-TDRS), etc
- [x] delete _ACFTITAN_SFTP_PYTEST_ environment variable
- [x] delete all sftp_config environment variables (ACF_TITAN_XYZ) in settings
- [x] delete python _pytest_sftpserver_ package and  update the pipfile lock
- [ ] Delete ACF Titan keys
- [x] Delete the line in secret_key_rotation_steps.py for _ACF Titan server keys_
- [x] Delete ACFTITAN keys from deploy-backend.sh
- [ ] Delete related env vars from backend and frontend docker-compose files   
- [x] Run Testing Checklist and confirm all tests pass
- [x] Test Case: Upload file and check to see if it fails 

**Notes:**

- Discussion with Alex and the dev team, we have decided to keep the LegacyFileTransfer model/table and create a follow up ticket

**Open Questions:**

- When should we delete the LegacyFileTransfer model?

### PR DESCRIPTION
**Pull request by raftmsohani**
## Summary of Changes

Pull request closes #2991 _

PR removes all logic for transferring uploaded files to ACF TITAN via SFTP

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
1. Proceed with functional tests as described herein.
1. Test steps should be captured in the demo GIF(s) and/or screenshots below.
> *Demo GIF(s) and screenshots for testing procedure*

Screenshot 2024-05-31 at 11 41 37 AM

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] files submitted through TDP are not transferred to ACFTitan server.

sftpp2
sftpp1

+ [ ] file transfer env vars are no longer present in the codebase, circleci, or backend environments
sftpp3

+ [x] backend logs do not show evidence of file transfer attempts.

```shell=
 2024-06-06T12:42:32.28-0400 [APP/PROC/WEB/0] OUT 10.255.141.85 - - [06/Jun/2024:16:42:32 +0000] "GET /v1/stts/alpha HTTP/1.0" 200 36455 "https://tdp-frontend-qasp.app.cloud.gov/data-files" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36"
   2024-06-06T12:42:34.58-0400 [APP/PROC/WEB/0] ERR [2024-06-06 16:42:34 +0000] [569] [DEBUG] GET /v1/data_files/
   2024-06-06T12:42:34.63-0400 [APP/PROC/WEB/0] OUT 10.255.141.85 - - [06/Jun/2024:16:42:34 +0000] "GET /v1/data_files/?year=2024&quarter=Q1&stt=1&file_type=tanf HTTP/1.0" 200 740 "https://tdp-frontend-qasp.app.cloud.gov/data-files" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36"
   2024-06-06T12:42:37.47-0400 [APP/PROC/WEB/0] ERR [2024-06-06 16:42:37 +0000] [566] [DEBUG] Closing connection.
   2024-06-06T12:42:38.26-0400 [APP/PROC/WEB/0] ERR [2024-06-06 16:42:38 +0000] [568] [DEBUG] POST /v1/data_files/
   2024-06-06T12:42:38.30-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,309 DEBUG views.py::create:L57 :  DataFileViewSet: <rest_framework.request.Request: POST '/v1/data_files/'>
   2024-06-06T12:42:38.30-0400 [APP/PROC/WEB/0] ERR DataFileViewSet: <rest_framework.request.Request: POST '/v1/data_files/'>
   2024-06-06T12:42:38.31-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,310 DEBUG validators.py::validate_file_infection:L54 :  CLAMAV_NEEDED noted as True, proceeding with scan.
   2024-06-06T12:42:38.31-0400 [APP/PROC/WEB/0] ERR CLAMAV_NEEDED noted as True, proceeding with scan.
   2024-06-06T12:42:38.31-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,310 DEBUG clients.py::__init__:L37 :  Set clamav endpoint_url as 'http://tdp-clamav-nginx-dev.apps.internal:9000/scan'
   2024-06-06T12:42:38.31-0400 [APP/PROC/WEB/0] ERR Set clamav endpoint_url as 'http://tdp-clamav-nginx-dev.apps.internal:9000/scan'
   2024-06-06T12:42:38.31-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,311 DEBUG clients.py::scan_file:L64 :  Initiating virus scan for file: test_staging.txt
   2024-06-06T12:42:38.31-0400 [APP/PROC/WEB/0] ERR Initiating virus scan for file: test_staging.txt
   2024-06-06T12:42:38.32-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,323 DEBUG clients.py::scan_file:L90 :  File scan marked as CLEAN for file: test_staging.txt
   2024-06-06T12:42:38.32-0400 [APP/PROC/WEB/0] ERR File scan marked as CLEAN for file: test_staging.txt
   2024-06-06T12:42:38.61-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,613 DEBUG views.py::create:L64 :  DataFileViewSet: status: 201
   2024-06-06T12:42:38.61-0400 [APP/PROC/WEB/0] ERR DataFileViewSet: status: 201
   2024-06-06T12:42:38.61-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,617 INFO views.py::create:L69 :  Preparing parse task: User META -> user: alexandra.pennington@acf.hhs.gov, stt: Alabama (01). Datafile META -> datafile: 642, section: Active Case Data, quarter Q1, year 2024.
   2024-06-06T12:42:38.61-0400 [APP/PROC/WEB/0] ERR Preparing parse task: User META -> user: alexandra.pennington@acf.hhs.gov, stt: Alabama (01). Datafile META -> datafile: 642, section: Active Case Data, quarter Q1, year 2024.
   2024-06-06T12:42:38.65-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,655 INFO views.py::create:L74 :  Submitted parse task to queue for datafile 642.
   2024-06-06T12:42:38.65-0400 [APP/PROC/WEB/0] ERR Submitted parse task to queue for datafile 642.
   2024-06-06T12:42:38.68-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,689 INFO parser_task.py::parse:L25 :  DataFile parsing started for file ADS.E2J.NDM1.TS01
   2024-06-06T12:42:38.68-0400 [APP/PROC/WEB/0] ERR [2024-06-06 16:42:38,689: INFO/ForkPoolWorker-1] DataFile parsing started for file ADS.E2J.NDM1.TS01
   2024-06-06T12:42:38.82-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,822 DEBUG views.py::create:L83 :  DataFileViewSet: return val: <Response status_code=201, "text/html; charset=utf-8">
   2024-06-06T12:42:38.82-0400 [APP/PROC/WEB/0] ERR DataFileViewSet: return val: <Response status_code=201, "text/html; charset=utf-8">
   2024-06-06T12:42:38.82-0400 [APP/PROC/WEB/0] OUT 10.255.141.85 - - [06/Jun/2024:16:42:38 +0000] "POST /v1/data_files/ HTTP/1.0" 201 430 "https://tdp-frontend-qasp.app.cloud.gov/data-files" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36"
   2024-06-06T12:42:38.83-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,834 INFO row_schema.py::parse_and_validate:L60 :  2 preparser error(s) encountered.
   2024-06-06T12:42:38.83-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,834 INFO parse.py::parse_datafile:L34 :  Preparser Error: 2 header errors encountered.
   2024-06-06T12:42:38.83-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,834 DEBUG parse.py::bulk_create_errors:L150 :  Bulk creating ParserErrors.
   2024-06-06T12:42:38.83-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,839 INFO parse.py::bulk_create_errors:L153 :  Created 2/1 ParserErrors.
   2024-06-06T12:42:38.85-0400 [APP/PROC/WEB/0] ERR 2024-06-06 16:42:38,850 INFO parser_task.py::parse:L38 :  Parsing finished for file -> {id: 642, filename: test_staging.txt, STT: Alabama (01), S3 location: data_files/2024/Q1/1/Active Case Data/test_staging.txt} with status Rejected and 1 errors.
   2024-06-06T12:42:38.85-0400 [APP/PROC/WEB/0] ERR [2024-06-06 16:42:38,850: INFO/ForkPoolWorker-1] Parsing finished for file -> {id: 642, filename: test_staging.txt, STT: Alabama (01), S3 location: data_files/2024/Q1/1/Active Case Data/test_staging.txt} with status Rejected and 1 errors.
   2024-06-06T12:42:38.92-0400 [APP/PROC/WEB/0] ERR [2024-06-06 16:42:38 +0000] [567] [DEBUG] POST /v1/logs/
   2024-06-06T12:42:38.96-0400 [APP/PROC/WEB/0] OUT 10.255.141.85 - - [06/Jun/2024:16:42:38 +0000] "POST /v1/logs/ HTTP/1.0" 200 9 "https://tdp-frontend-qasp.app.clo
```
+ [x] all other existing TDP features are functioning as expected.
+ [x] Testing Checklist has been run and all tests pass
+ [ ] README is updated, if necessary
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: docs/Security-Compliance/File-Transfer-TDRS/README.md
Comment (elipe17) [inline docs/Security-Compliance/File-Transfer-TDRS/README.md:1]: Just a process question. Should we retain this type of documentation as a "history" or do we just want to let Git manage the "history"?

**Review by elipe17 — APPROVED** [review]
No transfer occurred when submitting datafiles as expected. Approving with the added comment that we still need to handle the env vars during this PRs merge.

**Review by jtimpe — APPROVED** [review]
image

my favorite kind of pr. good work!

File: docs/Security-Compliance/File-Transfer-TDRS/README.md
Comment (raftmsohani) [reply docs/Security-Compliance/File-Transfer-TDRS/README.md:1]: I decided to let git manage that history.

**Review by ADPennington — APPROVED** [review]
lgtm @raftmsohani 🥇 

- please delete the env vars from raft's circleci.  I have deleted the env vars from qasp env and will also plan to delete the env vars from hhs circleci and staging and prod apps in the next release.

**Pull request merged**